Foundation/src/ProcessRunner.cpp

```diff
@@ -0,0 +1,227 @@
+//
+// ProcessRunner.cpp
+//
+// Library: Foundation
+// Package: Processes
+// Module:  ProcessRunner
+//
+// Copyright (c) 2023, Applied Informatics Software Engineering GmbH.
+// Aleph ONE Software Engineering d.o.o.,
+// and Contributors.
+//
+// SPDX-License-Identifier:    BSL-1.0
+//
+
+
+#include "Poco/ProcessRunner.h"
+#include "Poco/PIDFile.h"
+#include "Poco/FileStream.h"
+#include "Poco/AutoPtr.h"
+#include "Poco/File.h"
+#include "Poco/Path.h"
+#include "Poco/String.h"
+#include <fstream>
+
+
+using Poco::Thread;
+using Poco::Process;
+using Poco::ProcessHandle;
+using Poco::FileInputStream;
+using Poco::AutoPtr;
+using Poco::File;
+using Poco::Path;
+using Poco::Stopwatch;
+
+
+namespace Poco {
+
+
+ProcessRunner::ProcessRunner(const std::string& cmd,
+		const Args& args,
+		const std::string& pidFile,
+		int options,
+		int timeout,
+		bool startProcess,
+		const Args& pidArgFmt): _cmd(cmd),
+			_args(args),
+			_pid(INVALID_PID),
+			_pidFile(pidFile),
+			_options(options),
+			_timeout(timeout),
+			_pPH(nullptr),
+			_started(false),
+			_logger(Poco::Logger::get("ProcessRunner"))
```

Comment:
remove Logger, typically framework classes don't have logger, we communicate important things through exceptions

Reply:
Fixed in [ca79ba8](https://github.com/pocoproject/poco/pull/4225/commits/ca79ba8cb058dba942dd9b86e912037325339d6e)